opc/resource_database_service_instance_test.go

```diff
@@ -0,0 +1,98 @@
+package opc
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/go-oracle-terraform/database"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccOPCDatabaseServiceInstance_Basic(t *testing.T) {
+	ri := acctest.RandInt()
+	config := testAccDatabaseServiceInstanceBasic(ri)
+	resourceName := "opc_database_service_instance.test"
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDatabaseServiceInstanceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDatabaseServiceInstanceExists,
+					resource.TestCheckResourceAttr(
+						resourceName, "description", "test service instance"),
+					resource.TestCheckResourceAttr(
+						resourceName, "edition", "EE_EP"),
+					resource.TestCheckResourceAttr(
+						resourceName, "version", "12.2.0.1"),
+					resource.TestCheckResourceAttr(
+						resourceName, "parameter.#", "1"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckDatabaseServiceInstanceExists(s *terraform.State) error {
+	client := testAccProvider.Meta().(*OPCClient).databaseClient.ServiceInstanceClient()
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "opc_database_service_instance" {
+			continue
+		}
+
+		input := database.GetServiceInstanceInput{
+			Name: rs.Primary.Attributes["name"],
+		}
+		if _, err := client.GetServiceInstance(&input); err != nil {
+			return fmt.Errorf("Error retrieving state of DatabaseServiceInstance %s: %s", input.Name, err)
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckDatabaseServiceInstanceDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*OPCClient).databaseClient.ServiceInstanceClient()
+	if client == nil {
+		return fmt.Errorf("Database Client is not initialized. Make sure to use `database_endpoint` variable or `OPC_DATABASE_ENDPOINT` env variable")
+	}
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "opc_database_service_instance" {
+			continue
+		}
+
+		input := database.GetServiceInstanceInput{
+			Name: rs.Primary.Attributes["name"],
+		}
+		if info, err := client.GetServiceInstance(&input); err == nil {
+			return fmt.Errorf("DatabaseServiceInstance %s still exists: %#v", input.Name, info)
+		}
```

Comment:
We are checking to see if the instance is still there after deletion. If there is no error then we found the instance which is a test failure. Should be good as is